File: 01_countLetters/countLetters.js
/** @format */

// Case inSensitive

const CountLetters = (str, letter) => {
  let count = 0;
  let string = str.toLowerCase();
  let letterInLowercase = letter.toLowerCase();

  // Strings have an length arrays

  for (let i = 0; i < string.length; i++) {
    if (string[i] === letterInLowercase) {
      count++;
    }
  }
  return `Letter ${letter} is present ${count} times in ${str}`;
};

console.log(CountLetters("My Name", "M"));
console.log(CountLetters("My Name", "a"));
console.log(CountLetters("My Name", "m"));

// Case Sensitive

/*

const CountLetters = (str, letter) => {
  let count = 0;
  let string = str;

  for (let i = 0; i < string.length; i++) {
    if (string[i] === letter) {
      count++;
    }
  }
  return `Letter ${letter} is present ${count} times in ${str}`;
};

console.log(CountLetters("My Name", "M"));
console.log(CountLetters("My Name", "a"));
console.log(CountLetters("My Name", "m"));

*/

File: 00_Basic_Challanges/01_countLetters/countLetters.py
# ? Case inSensitive

def countLetters (str, letter):

   count = 0
   string = str.lower()
   letterInLowercase = letter.lower()

   for i in range(len(string)):
        if string[i].lower() == letter:
            count += 1
            
            return f"Letter {letter} is present {count} times in {string}"



    
print(countLetters("My Name", "M"))
